<launch>
  
   <node name="keyboard_node" pkg="keyboard" type="keyboard" output="screen" />

   <node name="setpoints_node" pkg="controller" type="setpoints" output="screen" >
      <remap from="keyup" to="/keyboard_node/keyup" />
   </node>

   <node pkg="vicon_bridge" type="vicon_bridge" name="vicon" output="screen">
      <param name="stream_mode" value="ClientPull" type="str" />
      <param name="datastream_hostport" value="192.168.10.1:801" type="str" />
      <param name="tf_ref_frame_id" value="/world" type="str" />
   </node>

   <node name="position_node" pkg="controller" type="position" output="screen" >
      <param name="Kr1" value="1.6532" />
      <param name="Kr2" value="1.5982" />
      <param name="Kr3" value="2.5763" />
      <param name="Kv1" value="1.6532" />
      <param name="Kv2" value="1.5982" />
      <param name="Kv3" value="2.5763" />
		<param name="x_upper_limit" value="1.75" />
      <param name="x_lower_limit" value="-1.75" />
		<param name="y_upper_limit" value="1.75" />
      <param name="y_lower_limit" value="-1.75" />
      <param name="frequency" value="20" />
      <remap from="state" to="/vicon/odro/odro" />
      <remap from="position_enable" to="/general_enable" /> 
   </node>

   <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" />

</launch>
